.circleci/circle-ci-config.py

```diff
@@ -0,0 +1,196 @@
+            '_default_job_':                        {'parallelism': 50,    'exec_resource_class': 'large'},
+            'jobs': {
+                # large 10
+                'j8_jvm_dtests':                    {'parallelism': 10},
+                'j8_jvm_dtests_vnode':              {'parallelism': 10},
+                # large 4
+                'j8_jvm_upgrade_dtests':            {'parallelism': 4},
+                # medium 10
+                'j11_jvm_dtests':                   {'parallelism': 10, 'exec_resource_class': 'medium'},
+                'j11_jvm_dtests_vnode':                   {'parallelism': 10, 'exec_resource_class': 'medium'},
+
+                # xlarge 100
+                'j8_upgradetests-no-vnodes':        {'parallelism': 100, 'exec_resource_class': 'xlarge'},
+
+                # xlarge 25
+                'repeated_upgrade_dtest':           {'parallelism': 25, 'exec_resource_class': 'xlarge'},
+
+                # large 25
+                'j11_repeated_dtest':               {'parallelism': 25},
+                'j8_repeated_dtest':                {'parallelism': 25},
+                'repeated_jvm_upgrade_dtest':       {'parallelism': 25},
+
+                # medium 25
+                'j11_repeated_utest':               {'parallelism': 25, 'exec_resource_class': 'medium'},
+                'j11_unit_tests':                   {'parallelism': 25, 'exec_resource_class': 'medium'},
+                'j8_repeated_utest':                {'parallelism': 25, 'exec_resource_class': 'medium'},
+                'j8_unit_tests':                    {'parallelism': 25, 'exec_resource_class': 'medium'},
+                'utests_compression':               {'parallelism': 25, 'exec_resource_class': 'medium'},
+                'utests_system_keyspace_directory': {'parallelism': 25, 'exec_resource_class': 'medium'},
+
+                'utests_fqltool':                       {'parallelism': 1, 'exec_resource_class': 'medium'},
+                'utests_long':                          {'parallelism': 1, 'exec_resource_class': 'medium'},
+                'utests_stress':                        {'parallelism': 1, 'exec_resource_class': 'medium'},
+            },
+        },
+        'HIGHER': {
+            # apply defaults to all jobs, so the below just overrides jobs which do not match the default
+            '_default_job_':                            {'parallelism': 100,    'exec_resource_class': 'xlarge'},
+            'jobs': {
+                'j8_jvm_upgrade_dtests':                {'parallelism': 2},
+                'j11_jvm_dtests':                       {'parallelism': 5},
+                'j11_jvm_dtests_vnode':                 {'parallelism': 5},
+                'j8_jvm_dtests':                        {'parallelism': 5},
+                'j8_jvm_dtests_vnode':                  {'parallelism': 5},
+
+                'utests_fqltool':                       {'parallelism': 1},
```

Comment:
I assume its an oversight that these use xlarge when they complete quickly with medium...